include/picongpu/plugins/radiation/Radiation.hpp

```diff
@@ -780,26 +783,104 @@ namespace picongpu
-                        {::openPMD::UnitDimension::L, 2.0},
-                        {::openPMD::UnitDimension::M, 1.0},
-                        {::openPMD::UnitDimension::T, -1.0}});
-
-                    /* get the radiation amplitude unit */
-                    Amplitude UnityAmplitude(1., 0., 0., 0., 0., 0.);
-                    const picongpu::float_64 factor = UnityAmplitude.calc_radiation() * UNIT_ENERGY * UNIT_TIME;
+                    // begin: write per-rank amplitude data
+                    if(writeDistributedAmplitudes)
+                    {
+                        ::openPMD::Mesh mesh_amp = openPMDdataFileIteration.meshes["Amplitude_distributed"];
+
+                        mesh_amp.setGeometry(::openPMD::Mesh::Geometry::cartesian); // set be default
+                        mesh_amp.setDataOrder(::openPMD::Mesh::DataOrder::C);
+                        mesh_amp.setGridSpacing(std::vector<double>{1.0, 1.0, 1.0});
+                        mesh_amp.setGridGlobalOffset(std::vector<double>{0.0, 0.0, 0.0});
+                        mesh_amp.setGridUnitSI(1.0);
+                        mesh_amp.setAxisLabels(
+                            std::vector<std::string>{"MPI", "detector direction index", "detector frequency index"});
+                        mesh_amp.setUnitDimension(std::map<::openPMD::UnitDimension, double>{
+                            {::openPMD::UnitDimension::L, 2.0},
+                            {::openPMD::UnitDimension::M, 1.0},
+                            {::openPMD::UnitDimension::T, -1.0}});
+
+                        /* get the radiation amplitude unit */
+                        Amplitude UnityAmplitude(1., 0., 0., 0., 0., 0.);
+                        const picongpu::float_64 factor = UnityAmplitude.calc_radiation() * UNIT_ENERGY * UNIT_TIME;
+
+                        // buffer for data re-arangement
+                        const int N_tmpBuffer = radiation_frequencies::N_omega * parameters::N_observer;
+                        std::vector<std::complex<float_64>> fallbackBuffer;
+
+                        // reshape abstract MeshRecordComponent
+                        ::openPMD::Datatype datatype_amp = ::openPMD::determineDatatype<std::complex<float_64>>();
+                        auto communicator = Environment<simDim>::get().GridController().getCommunicator();
+                        ::openPMD::Extent total_extent_amp
+                            = {size_t(communicator.getSize()), parameters::N_observer, radiation_frequencies::N_omega};
+                        ::openPMD::Offset local_offset_amp = {size_t(communicator.getRank()), 0, 0};
+                        ::openPMD::Extent local_extent_amp
+                            = {1, parameters::N_observer, radiation_frequencies::N_omega};
+
+                        auto srcBuffer = radiation->getHostBuffer().getBasePointer();
+
+                        for(uint32_t ampIndex = 0; ampIndex < Amplitude::numComponents / 2; ++ampIndex)
```

Comment:
This is due to the changed representation of imaginary numbers in the distributed output (see output [here](https://github.com/ComputationalRadiationPhysics/picongpu/pull/4456#issuecomment-1497205476)).
In a 3D simulation, numComponents is 6, but we create only 3 components.

Reply:
Thanks for the explanation. That makes sense – could have seen that :sweat_smile:

Reply:
```suggestion
                        /*
                         * numComponents includes the components of a complex number, e.g. in a 3D simulation,
                         * numComponents is 6.
                         * Since the distributed output uses native complex types, we don't want this.
                         * ---> Amplitude::numComponents / 2
                         */
                        for(uint32_t ampIndex = 0; ampIndex < Amplitude::numComponents / 2; ++ampIndex)
```

Reply:
I've added an inline comment now.